compiler/compiler_test.go

```diff
@@ -41,27 +43,336 @@ func Bfunc() int {
+	sel := declSelection(t, srcFiles, nil)
+
+	sel.DeclCode.IsAlive(`^\s*Foo = \$newType`)
+	sel.DeclCode.IsAlive(`^\s*\$ptrType\(Foo\)\.prototype\.Bar`)
+
+	sel.DeclCode.IsDead(`^\s*\$ptrType\(Foo\)\.prototype\.baz`)
+}
+
+func TestDeclSelection_KeepUnusedUnexportedMethodForInterface(t *testing.T) {
+	src := `
+		package main
+		type Foo struct {}
+		func (f Foo) Bar() {
+			println("foo")
+		}
+		func (f Foo) baz() {} // unused
+
+		type Foo2 struct {}
+		func (f Foo2) Bar() {
+			println("foo2")
+		}
+
+ 		type IFoo interface {
+			Bar()
+			baz()
+		}
+		func main() {
+			fs := []any{ Foo{}, Foo2{} }
+			for _, f := range fs {
+				if i, ok := f.(IFoo); ok {
+					i.Bar()
+				}
+			}
+		}`
+
+	srcFiles := []source{{`main.go`, []byte(src)}}
+	sel := declSelection(t, srcFiles, nil)
+
+	sel.DeclCode.IsAlive(`^\s*Foo = \$newType`)
+	sel.DeclCode.IsAlive(`^\s*\$ptrType\(Foo\)\.prototype\.Bar`)
+
+	// `baz` is used to duck-type (via method list) against IFoo
+	// but the method itself is not used so can be removed.
+	sel.DeclCode.IsDead(`^\s*\$ptrType\(Foo\)\.prototype\.baz`)
+	sel.MethodListCode.IsAlive(`^\s*Foo.methods = .* \{prop: "baz", name: "baz"`)
+}
+
+func TestDeclSelection_KeepUnexportedMethodUsedViaInterfaceLit(t *testing.T) {
+	src := `
+		package main
+		type Foo struct {}
+		func (f Foo) Bar() {
+			println("foo")
+		}
+		func (f Foo) baz() {
+			println("baz")
+		}
+		func main() {
+			var f interface {
+				Bar()
+				baz()
+			} = Foo{}
+			f.baz()
+		}`
+
+	srcFiles := []source{{`main.go`, []byte(src)}}
+	sel := declSelection(t, srcFiles, nil)
+
+	sel.DeclCode.IsAlive(`^\s*Foo = \$newType`)
+	sel.DeclCode.IsAlive(`^\s*\$ptrType\(Foo\)\.prototype\.Bar`)
+	sel.DeclCode.IsAlive(`^\s*\$ptrType\(Foo\)\.prototype\.baz`)
+}
+
+func TestDeclSelection_KeepAliveUnexportedMethodsUsedInMethodExpressions(t *testing.T) {
+	src := `
+		package main
+		type Foo struct {}
+		func (f Foo) baz() {
+			println("baz")
+		}
+		func main() {
+			fb := Foo.baz
+			fb(Foo{})
+		}`
+
+	srcFiles := []source{{`main.go`, []byte(src)}}
+	sel := declSelection(t, srcFiles, nil)
+
+	sel.DeclCode.IsAlive(`^\s*Foo = \$newType`)
+	sel.DeclCode.IsAlive(`^\s*\$ptrType\(Foo\)\.prototype\.baz`)
+}
+
+func TestDeclSelection_RemoveUnusedFuncInstance(t *testing.T) {
+	src := `
+		package main
+		func Sum[T int | float64](values ...T) T {
+			var sum T
+			for _, v := range values {
+				sum += v
+			}
+			return sum
+		}
+		func Foo() { // unused
+			println(Sum(1, 2, 3))
+		}
+		func main() {
+			println(Sum(1.1, 2.2, 3.3))
+		}`
+
+	srcFiles := []source{{`main.go`, []byte(src)}}
+	sel := declSelection(t, srcFiles, nil)
+
+	sel.DeclCode.IsAlive(`^\s*Sum\[\d+ /\* float64 \*/\]`)
+	sel.DeclCode.IsAlive(`^\s*sliceType(\$\d+)? = \$sliceType\(\$Float64\)`)
+
+	sel.DeclCode.IsDead(`^\s*Foo = function`)
+	sel.DeclCode.IsDead(`^\s*sliceType(\$\d+)? = \$sliceType\(\$Int\)`)
+
+	// TODO(gn): This should not be alive because it is not used.
```

Comment:
I recommend using a full GitHub username in TODOs to make the author unambiguous. This is more important for longer-lived TODOs, where it may point at who to ask for context/clarifications if need be. But for the sake of consistency, I would do it here too.